source/ddbus/exception.d

```diff
@@ -27,6 +27,27 @@ class DBusException : Exception {
   }
 }
 
+/++
+  Throw this from a handler to have it automatically translate to an Error
+  reply.
++/
+class DBusErrorReturn : Exception {
+  string errorName;
+
```

Comment:
I think you should still add a default public constructor with just a message, especially with regards to custom memory allocation as there is only a helper function with `new`.

Reply:
But isn't the errorName a mandatory field in DBus errors? I can't imagine the C library will like it if you passed in an empty string for the error.

Hmm, I suppose, though, it could be set to some standard default error name which can be overridden by the user. Let me do that instead...

Reply:
yeah default with user overriding value sounds good. It's a public member so after construction it can always be changed.